### PR TITLE
albert rebased

### PR DESCRIPTION
**Pull request by StefanBuerscher**
Albert changes rebased.

### CODE REVIEW

**Comment by StefanBuerscher** [comment]
ok to test

**Comment by obusher** [comment]
Which branch did you rebase ?
PS: I assume you should target 'Catrod:feature/albert'.

File: catroid/src/org/catrobat/catroid/content/bricks/PointInDirectionBrick.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ private void initializeBrickFields(Formula direction) {
 	@Override
 	public int getRequiredResources() {
 		return NO_RESOURCES;
-	}
+	} //TODO: albert :if (getFormulaWithBrickField(BrickField.DEGREES).containsRobotAlbertSensors() == true) {
```
Comment (avihric) [inline catroid/src/org/catrobat/catroid/content/bricks/PointInDirectionBrick.java]: Is this a relic?

File: catroid/src/org/catrobat/catroid/content/bricks/RobotAlbertDistanceSensorLeftBrick.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+		final Brick brickInstance = this;
+		checkbox.setOnCheckedChangeListener(new OnCheckedChangeListener() {
+			@Override
+			public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+				checked = isChecked;
+				adapter.handleCheck(brickInstance, isChecked);
+			}
+		});
+
```
Comment (avihric) [inline catroid/src/org/catrobat/catroid/content/bricks/RobotAlbertDistanceSensorLeftBrick.java]: Why is the following commented code there?

File: catroid/src/org/catrobat/catroid/content/bricks/RobotAlbertDistanceSensorRightBrick.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+		setCheckboxView(R.id.brick_robot_albert_distance_sensor_right_checkbox);
+
+		final Brick brickInstance = this;
+		checkbox.setOnCheckedChangeListener(new OnCheckedChangeListener() {
+			@Override
+			public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+				checked = isChecked;
+				adapter.handleCheck(brickInstance, isChecked);
+			}
+		});
```
Comment (avihric) [inline catroid/src/org/catrobat/catroid/content/bricks/RobotAlbertDistanceSensorRightBrick.java]: same here?

File: catroid/src/org/catrobat/catroid/content/bricks/SetVariableBrick.java [hunk 4]
```diff
@@ -70,6 +75,13 @@ public SetVariableBrick(double value) {
 		this.userVariable = null;
 		initializeBrickFields(new Formula(value));
 	}
+	//TODO: albert
```
Comment (avihric) [inline catroid/src/org/catrobat/catroid/content/bricks/SetVariableBrick.java]: remove TODO

File: catroid/src/org/catrobat/catroid/content/bricks/SetVariableBrick.java [hunk 5]
```diff
@@ -202,6 +214,21 @@ public View getPrototypeView(Context context) {
 
 		TextView textSetVariable = (TextView) prototypeView.findViewById(R.id.brick_set_variable_prototype_view);
 		textSetVariable.setText(String.valueOf(BrickValues.SET_VARIABLE));
+		//TODO: albert
```
Comment (avihric) [inline catroid/src/org/catrobat/catroid/content/bricks/SetVariableBrick.java]: remove TODO

File: catroid/src/org/catrobat/catroid/devices/albert/SensorRobotAlbert.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+import org.catrobat.catroid.formulaeditor.SensorCustomEvent;
+import org.catrobat.catroid.formulaeditor.SensorCustomEventListener;
+import org.catrobat.catroid.formulaeditor.Sensors;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class SensorRobotAlbert {
+
+	//TODO: albert include albert sensor in formula editor
```
Comment (avihric) [inline catroid/src/org/catrobat/catroid/devices/albert/SensorRobotAlbert.java]: Is this TODO still valid?

File: catroid/src/org/catrobat/catroid/formulaeditor/SensorHandler.java [hunk 7]
```diff
@@ -71,6 +77,16 @@ public static void startSensorListener(Context context) {
 		instance.sensorManager.registerListener(instance, Sensors.LOUDNESS);
 		FaceDetectionHandler.registerOnFaceDetectedListener(instance);
 		FaceDetectionHandler.registerOnFaceDetectionStatusListener(instance);
+		//TODO: albert Tobi fragen was sharedPref. macht
```
Comment (avihric) [inline catroid/src/org/catrobat/catroid/formulaeditor/SensorHandler.java]: Has Tobi been asked? :)

File: catroid/src/org/catrobat/catroid/stage/PreStageActivity.java [hunk 8]
```diff
@@ -251,9 +279,10 @@ public static void shutdownResources() {
 		if (legoNXT != null) {
 			legoNXT.pauseCommunicator();
 		}
-        if (FaceDetectionHandler.isFaceDetectionRunning()) {
-            FaceDetectionHandler.stopFaceDetection();
-        }
+//		ServiceProvider.getService(CatroidService.BLUETOOTH_DEVICE_SERVICE).pause();
```
Comment (avihric) [inline catroid/src/org/catrobat/catroid/stage/PreStageActivity.java]: Remove comment

File: catroid/src/org/catrobat/catroid/ui/fragment/CategoryBricksFactory.java [hunk 9]
```diff
@@ -322,6 +330,23 @@
...
+		List<Brick> robotAlbertBrickList = new ArrayList<Brick>();
+		robotAlbertBrickList.add(new RobotAlbertMotorBrick( RobotAlbertMotorBrick.Motor.Left,
+				BrickValues.ROBOT_ALBERT_SPEED));
+		robotAlbertBrickList.add(new RobotAlbertBuzzerBrick( BrickValues.ROBOT_ALBERT_BUZZER));
+		robotAlbertBrickList.add(new RobotAlbertRgbLedEyeBrick(RobotAlbertRgbLedEyeBrick.Eye.Both,
+				BrickValues.ROBOT_ALBERT_RGB_LED_EYE_RED, BrickValues.ROBOT_ALBERT_RGB_LED_EYE_GREEN,
+				BrickValues.ROBOT_ALBERT_RGB_LED_EYE_BLUE));
+		robotAlbertBrickList.add(new RobotAlbertFrontLedBrick( BrickValues.ROBOT_ALBERT_FRONT_LED));
+		robotAlbertBrickList.add(new RobotAlbertBodyLedBrick( BrickValues.ROBOT_ALBERT_BODY_LED));
+		//TODO: albert
```
Comment (avihric) [inline catroid/src/org/catrobat/catroid/ui/fragment/CategoryBricksFactory.java]: remove TODO

File: catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorListFragment.java [hunk 10]
```diff
@@ -133,7 +145,14 @@ public void onCreate(Bundle savedInstanceState) {
 		} else if (tag.equals(LOGIC_TAG)) {
 			itemsIds = LOGIC_ITEMS;
 		} else if (tag.equals(SENSOR_TAG)) {
-			itemsIds = SENSOR_ITEMS;
+			//TODO ALbert:
```
Comment (avihric) [inline catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorListFragment.java]: remove TODO

File: catroidTest/src/org/catrobat/catroid/uitest/content/brick/BrickValueParameterTest.java [hunk 11]
```diff
@@ -437,6 +450,78 @@ public void testLegoBricksDefaultValues() {
 				nxtPlayToneEditTextValue);
 	}
 
+	public void testRobotAlbertBricksDefaultValues() {
+
+		//TODO: Check if Spinner-Value is correct
```
Comment (avihric) [inline catroidTest/src/org/catrobat/catroid/uitest/content/brick/BrickValueParameterTest.java]: remove TODO

File: catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorListFragment.java (hunk 10 above)
Comment (StefanBuerscher) [inline catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorListFragment.java]: Wie gesagt, ich hab von diesem Code nichts programmiert, aber ich werde 
die Mängel beheben.
Bis wann soll das erfolgen?

lg Stefan

File: catroidTest/src/org/catrobat/catroid/uitest/content/brick/RobotAlbertBuzzerBrickTest.java [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+		assertEquals("Wrong Brick instance.", projectBrickList.get(0), adapter.getChild(groupCount - 1, 0));
+		assertNotNull("TextView does not exist.",
+				solo.getText(solo.getString(R.string.brick_robot_albert_buzzer_action)));
+
+		assertNotNull("TextView does not exist.", solo.getText(solo.getString(R.string.robot_albert_buzzer_frequency)));
+
+		UiTestUtils.testBrickWithFormulaEditor(solo, ProjectManager.getInstance().getCurrentSprite(), R.id.robot_albert_buzzer_frequency_edit_text, SET_FREQ,
+				Brick.BrickField.ROBOT_ALBERT_BUZZER, brick);
+
+		//TODO: Prints an error after setting it from init to new value, but I can see that it was set. Executing the command a second time seams to solve that. Why not the first time??
```
Comment (avihric) [inline catroidTest/src/org/catrobat/catroid/uitest/content/brick/RobotAlbertBuzzerBrickTest.java]: Is this resolved?

File: catroidTest/src/org/catrobat/catroid/uitest/content/brick/RobotAlbertMotorActionBrickTest.java [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+
+		String[] motors = getActivity().getResources().getStringArray(R.array.robot_albert_motor_chooser);
+		assertTrue("Spinner items list too short!", motors.length == 3);
+
+		int spinnerIndex = 1;
+
+		if (Build.VERSION.SDK_INT < Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
+			spinnerIndex = 0;
+		}
+		//TODO: seams that the if above isn't true......
```
Comment (avihric) [inline catroidTest/src/org/catrobat/catroid/uitest/content/brick/RobotAlbertMotorActionBrickTest.java]: remove TODO

File: catroidTest/src/org/catrobat/catroid/uitest/content/brick/RobotAlbertRgbLedEyeActionBrickTest.java [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+
+		String[] eyes = getActivity().getResources().getStringArray(R.array.robot_albert_eye_chooser);
+		assertTrue("Spinner items list too short!", eyes.length == 3);
+
+		int spinnerIndex = 1;
+
+		if (Build.VERSION.SDK_INT < Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
+			spinnerIndex = 0;
+		}
+		//TODO: seams that the if above isn't true......
```
Comment (avihric) [inline catroidTest/src/org/catrobat/catroid/uitest/content/brick/RobotAlbertRgbLedEyeActionBrickTest.java]: remove TODO

File: catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorListFragment.java (hunk 10 above)
Comment (avihric) [inline catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorListFragment.java]: There is no hurry, currently the release is being prepared.
Btw please write in english.
Regards

**Comment by avihric** [comment]
PS: @obusher is right, the PR should point to feature/albert

File: catroid/res/layout/brick_robot_albert_body_led_action.xml [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+            <TextView
+                android:id="@+id/robot_albert_body_led_status_text_view"
+                style="@style/BrickText"
+                app:layout_newLine="true"
+                android:text="@string/robot_albert_body_led_status" >
+            </TextView>
+            
+            <TextView
+                android:id="@+id/robot_albert_body_led_status_prototype_text_view"
+                style="@style/BrickPrototypeTextView" />
```
Comment (aried3r) [inline catroid/res/layout/brick_robot_albert_body_led_action.xml]: Please stick with one style of using these elements, either self closing `<TextView />` (which I think the rest of the layouts use) or with an end tag `<TextView></TextView>`.

And please run the formatter on the files, some of them are wrongly indented.

File: catroidTest/src/org/catrobat/catroid/uitest/content/brick/RobotAlbertMotorActionBrickTest.java [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+
+
+		String[] motors = getActivity().getResources().getStringArray(R.array.robot_albert_motor_chooser);
+		assertTrue("Spinner items list too short!", motors.length == 3);
+
+		int spinnerIndex = 1;
+
+		if (Build.VERSION.SDK_INT < Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
+			spinnerIndex = 0;
+		}
```
Comment (aried3r) [inline catroidTest/src/org/catrobat/catroid/uitest/content/brick/RobotAlbertMotorActionBrickTest.java]: Not needed anymore, we're using `minSdkVersion = 16` now.

File: catroidTest/src/org/catrobat/catroid/uitest/content/brick/RobotAlbertRgbLedEyeActionBrickTest.java [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+				R.id.robot_albert_rgb_led_action_blue_edit_text, SET_BLUE_STRING, Brick.BrickField.ROBOT_ALBERT_RGB_BLUE, brick);
+
+		String[] eyes = getActivity().getResources().getStringArray(R.array.robot_albert_eye_chooser);
+		assertTrue("Spinner items list too short!", eyes.length == 3);
+
+		int spinnerIndex = 1;
+
+		if (Build.VERSION.SDK_INT < Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
+			spinnerIndex = 0;
+		}
```
Comment (aried3r) [inline catroidTest/src/org/catrobat/catroid/uitest/content/brick/RobotAlbertRgbLedEyeActionBrickTest.java]: Same here.

**Comment by obusher** [comment]
Before you start with changing the code, i want that you rebase the old branch again.
You did not write the code so why did you squash and took all the commits ? :pager:

**Comment by StefanBuerscher** [comment]
took all the commits?

im not very happy with this situation. I did not come to this project to 
do all the boring stuff that needs to be done and nothing new. But I 
will finish it anyway.

So what do you mean with "You did not write the code so why did you 
squash and took all the commits ?" ?

**Comment by aried3r** [comment]
> But I will finish it anyway.

And we appreciate it! :)

**Comment by obusher** [comment]
There exists a proper way to `rebase` a branch of someone else.
The common way is only to run `git rebase` and preserve the `git-commit-tree` (commit history) of the guy who wrote the code. If the `git-commit-tree` is not very beautiful, you can [rewrite the history](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase-i) (f.e. squash) of some commits but in such way the code owner does not loose anything.

Personally i would not squash all commits with 5k+ lines to a single one. 
However you should not own the code of someone else.

After successful `rebase` you can commit your changes mentioned above in a normal way.
(Please link the root branch in your PR.) :taxi:

**Comment by avihric** [comment]
Please open a new PR pointing to `feature/albert`

**Comment by StefanBuerscher** [comment]
Ok but I dont have time till monday

**Pull request closed without merging**